### PR TITLE
chall12 - gbudau

### PR DESCRIPTION
**Pull request by gbudau**
**How i solved this problem**  /  **Short description of the code**
1. Check if all open parentheses have an closing pair, then copy the string on the heap using `strdup()`.
2. Save the indexes after an open parenthesis on a `stack` array.
3. Loop over the copy and if the current character is an open parenthesis save the `index + 1` on the `stack`.
4. If the current character is a closing parenthesis first invert all parentheses inside the current parentheses then reverse the string inside the parentheses.
5. Free the `stack` and return the modified string.

### CODE REVIEW

File: chall12/gbudau.c [hunk 1]
```diff
@@ -0,0 +1,117 @@
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+static bool		valid_parenthesis(const char *str)
+{
+	size_t	open;
```
Comment (clafoutis42) [inline chall12/gbudau.c:7]: It is bad practice to name a variable as `open` (as `open` is meant to open files)

File: chall12/gbudau.c [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+	open = false;
```
Comment (clafoutis42) [inline chall12/gbudau.c:9]: Here you are using `open` as a counter, assigning it `false` is confusing.

File: chall12/gbudau.c [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+{
+	while (start <= end)
+	{
+		if (str[start] == ')')
+			str[start] = '(';
+		else if (str[start] == '(')
+			str[start] = ')';
+		start++;
+	}
+}
```
Comment (clafoutis42) [inline chall12/gbudau.c:77]: You could do this while copying instead of going through you string all over again

**Review by clafoutis42 — COMMENTED** [review]
Your algorithm is correct, congrats you seem to pass all the tests !  
However you could have optimized a little by reversing while reading, but this is really just optimization.

File: chall12/gbudau.c (hunk 1 above)
Comment (gbudau) [reply chall12/gbudau.c:7]: My bad, didn't think about it.

File: chall12/gbudau.c (hunk 3 above)
Comment (gbudau) [reply chall12/gbudau.c:77]: Thank you, will keep it in mind for next time.

**Comment by gbudau** [comment]
Thank you Charles! I will try to optimize more in the future :+1: .

**Pull request merged**